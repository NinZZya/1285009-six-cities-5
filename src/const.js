export const AppPath = {
  ROOT: `/`,
  LOGIN: `/login`,
  CITY: `/city`,
  OFFER: `/offer`,
  FAVORITES: `/favorites`,
  NOT_FOUND: `/not-found`,
};

export const CITIES = {
  1: {id: 1, name: `Paris`},
  2: {id: 2, name: `Cologne`},
  3: {id: 3, name: `Brussels`},
  4: {id: 4, name: `Amsterdam`},
  5: {id: 5, name: `Hamburg`},
  6: {id: 6, name: `Dusseldorf`},
};

export const SortType = {
  POPULAR: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const STARS = [
  {value: 5, title: `perfect`},
  {value: 4, title: `good`},
  {value: 3, title: `not bad`},
  {value: 2, title: `badly`},
  {value: 1, title: `terribly`},
];
