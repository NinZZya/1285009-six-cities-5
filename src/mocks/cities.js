export default {
  1: {
    id: 1,
    name: `Paris`,
    coords: [48.864716, 2.349014],
  },
  2: {
    id: 2,
    name: `Cologne`,
    coords: [50.9415491, 6.9408247],
  },
  3: {
    id: 3,
    name:
    `Brussels`,
    coords: [50.8503396, 4.3517103],
  },
  4: {
    id: 4,
    name: `Amsterdam`,
    coords: [52.38333, 4.9],
  },
  5: {id: 5, name: `Hamburg`},
  6: {id: 6, name: `Dusseldorf`},
};
