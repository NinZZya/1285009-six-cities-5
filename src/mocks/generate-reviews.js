import {
  getRandomArr,
  getRandomArrValue,
  getRandomInt,
} from './randomaizer';

const DESCRIPTIONS = [
  `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
];

const USERS = [
  {id: 1, name: `Max`, avatar: `/img/avatar-max.jpg`},
  {id: 2, name: `Angelina`, avatar: `/img/avatar-angelina.jpg`},
  {id: 3, name: `John`, avatar: `/img/avatar.svg`},
  {id: 4, name: `Ann`, avatar: `/img/avatar.svg`},
];

let reviewId = 0;

const generateDate = () => {
  const year = getRandomInt(2018, 2020);
  const month = String(getRandomInt(1, 12)).padStart(2, `0`);
  const day = String(getRandomInt(1, 28)).padStart(2, `0`);
  return `${year}-${month}-${day}`;
};

const generateReview = () => {
  const user = getRandomArrValue(USERS);

  return {
    id: reviewId++,
    user,
    date: generateDate(),
    rate: getRandomInt(0, 5),
    text: getRandomArr(DESCRIPTIONS, getRandomInt(1, DESCRIPTIONS.length)).join(`/n`),
  };
};

const generateReviews = (offers) => offers.reduce((reviews, offer) => {
  reviews[offer.id] = new Array(getRandomInt(0, 5)).fill(``).map(generateReview);
  return reviews;
}, {});

export default generateReviews;
