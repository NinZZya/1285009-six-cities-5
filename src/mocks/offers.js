export default [
  {
    id: 0,
    city: {
      id: 1,
      name: `Paris`,
    },
    title: `Wood and stone place`,
    type: `Private room`,
    price: 228,
    rate: 0,
    bedroomsCount: 1,
    adultsCount: 1,
    features: [
      `Dishwasher`,
      `Baby seat`,
    ],
    host: {
      name: `Angelina`,
      avatar: `/img/avatar-angelina.jpg`,
      isPro: false,
    },
    images: [
      `/img/apartment-03.jpg`,
      `/img/apartment-02.jpg`,
    ],
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    reviews: [
      {
        id: 0,
        user: {
          id: 3,
          name: `John`,
          avatar: `/img/avatar.svg`,
        },
        date: `2019-11-26`,
        rate: 2,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      },
      {
        id: 1,
        user: {
          id: 3,
          name: `John`,
          avatar: `/img/avatar.svg`
        },
        date: `2019-03-23`,
        rate: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      }
    ],
    isPremium: false,
  },
  {
    id: 1,
    city: {
      id: 1,
      name: `Paris`,
    },
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    price: 97,
    rate: 3,
    bedroomsCount: 1,
    adultsCount: 3,
    features: [
      `Dishwasher`,
      `Towels`,
      `Fridge`,
      `Heating`,
      `Washing machine`,
      `Wi-Fi`,
      `Coffee machine`,
    ],
    host: {
      name: `John`,
      avatar: `/img/avatar.svg`,
      isPro: false,
    },
    images: [
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
    ],
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    reviews: [
      {
        id: 2,
        user: {
          id: 4,
          name: `Ann`,
          avatar: `/img/avatar.svg`
        },
        date: `2018-09-22`,
        rate: 2,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      },
      {
        id: 3,
        user: {
          id: 1,
          name: `Max`,
          avatar: `/img/avatar-max.jpg`
        },
        date: `2018-10-24`,
        rate: 0,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      },
      {
        id: 4,
        user: {
          id: 4,
          name: `Ann`,
          avatar: `/img/avatar.svg`
        },
        date: `2019-08-16`,
        rate: 1,
        text: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      }
    ],
    isPremium: false,
  },
  {
    id: 2,
    city: {
      id: 2,
      name: `Paris`
    },
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    price: 89,
    rate: 1,
    bedroomsCount: 2,
    adultsCount: 1,
    features: [],
    host: {
      name: `Angelina`,
      avatar: `/img/avatar-angelina.jpg`,
      isPro: false,
    },
    images: [
      `/img/apartment-01.jpg`
    ],
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    reviews: [
      {
        id: 5,
        user: {
          id: 4,
          name: `Ann`,
          avatar: `/img/avatar.svg`,
        },
        date: `2019-06-07`,
        rate: 3,
        text: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      },
      {
        id: 6,
        user: {
          id: 1,
          name: `Max`,
          avatar: `/img/avatar-max.jpg`,
        },
        date: `2019-06-02`,
        rate: 1,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      }
    ],
    isPremium: false,
  },
  {
    id: 3,
    city: {
      id: 2,
      name: `Paris`,
    },
    title: `Wood and stone place`,
    type: `Private room`,
    price: 381,
    rate: 3,
    bedroomsCount: 1,
    adultsCount: 1,
    features: [
      `Fridge`,
      `Towels`,
    ],
    host: {
      name: `Angelina`,
      avatar: `/img/avatar-angelina.jpg`,
      isPro: false
    },
    images: [
      `/img/room.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
    ],
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    reviews: [
      {
        id: 7,
        user: {
          id: 4,
          name: `Ann`,
          avatar: `/img/avatar.svg`,
        },
        date: `2018-11-08`,
        rate: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      },
      {
        id: 8,
        user: {
          id: 2,
          name: `Angelina`,
          avatar: `/img/avatar-angelina.jpg`
        },
        date: `2019-02-23`,
        rate: 0,
        text: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      }
    ],
    isPremium: false,
  }
];
