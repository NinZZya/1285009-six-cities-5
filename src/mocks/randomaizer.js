export const getRandomBool = () => Math.random > 0.5;
export const getRandomInt = (min, max) => min + Math.floor(Math.random() * (max - min));
export const getRandomArrValue = (arr) => arr[Math.floor(Math.random() * arr.length)];
export const getRandomArr = (arr, length = arr.length) => {
  const randomArr = arr.slice();
  randomArr.sort(() => Math.random() - 0.5);

  return randomArr.slice(0, length);
};
