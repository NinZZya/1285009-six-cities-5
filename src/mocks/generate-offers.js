import {CITIES} from '../const';
import {
  getRandomArr,
  getRandomArrValue,
  getRandomBool,
  getRandomInt,
} from './randomaizer';

const TITLES = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
  `Wood and stone place`,
];

const FEATURES = [
  `Wi-Fi`,
  `Washing machine`,
  `Towels`,
  `Heating`,
  `Coffee machine`,
  `Baby seat`,
  `Kitchen`,
  `Dishwasher`,
  `Cabel TV`,
  `Fridge`,
];

const DESCRIPTIONS = [
  `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
];

const IMAGES = [
  `/img/apartment-01.jpg`,
  `/img/apartment-02.jpg`,
  `/img/apartment-03.jpg`,
  `/img/room.jpg`,
];

const USERS = [
  {id: 1, name: `Max`, avatar: `/img/avatar-max.jpg`},
  {id: 2, name: `Angelina`, avatar: `/img/avatar-angelina.jpg`},
  {id: 3, name: `John`, avatar: `/img/avatar.svg`},
  {id: 4, name: `Ann`, avatar: `/img/avatar.svg`},
];

const OFFERS_COORDS = {
  1: [
    [48.8588336, 2.2457026],
    [48.884716, 2.339014],
    [48.8727168, 2.3359014],
    [48.8560346, 2.3181479],
    [48.8553232, 2.3522368],
    [48.8728688, 2.3559089],
    [48.8628688, 2.3359089],
  ],
  2: [
    [50.9422757, 6.9500156],
    [50.947828, 6.9184057],
    [50.9435302, 6.9600436],
    [50.9468833, 6.9609954],
    [50.9396526, 6.941616],
    [50.9386526, 6.917616],
  ],
  3: [
    [50.8488746, 4.3506942],
    [50.8559434, 4.3672311],
    [50.8513309, 4.3515856],
    [50.8414535, 4.3618321],
    [50.8434535, 4.3518321],
  ],
  4: [
    [52.3909553943508, 4.85309666406198],
    [52.369553943508, 4.85309666406198],
    [52.3909553943508, 4.929309666406198],
    [52.3809553943508, 4.939309666406198],
  ],
};

const TYPES = [`Apartment`, `Private room`];

const cities = Object.keys(CITIES).slice(0, length - 2);

let reviewId = 0;

const generateDate = () => {
  const year = getRandomInt(2018, 2020);
  const month = String(getRandomInt(1, 12)).padStart(2, `0`);
  const day = String(getRandomInt(1, 28)).padStart(2, `0`);
  return `${year}-${month}-${day}`;
};

const generateReview = () => {
  const user = getRandomArrValue(USERS);

  return {
    id: reviewId++,
    user,
    date: generateDate(),
    rate: getRandomInt(0, 5),
    text: getRandomArr(DESCRIPTIONS, getRandomInt(1, DESCRIPTIONS.length)).join(`/n`),
  };
};

const generateReviews = (count) => new Array(count).fill(``).map(generateReview);


const generateOffer = (cityId, id) => {
  const coords = getRandomArr(OFFERS_COORDS[cityId]);

  return coords.map((coord, index) => {
    const host = getRandomArrValue(USERS);

    return {
      id: id * 100 + index,
      city: CITIES[cityId],
      title: getRandomArrValue(TITLES),
      type: getRandomArrValue(TYPES),
      price: getRandomInt(20, 500),
      rate: getRandomInt(0, 5),
      bedroomsCount: getRandomInt(1, 3),
      adultsCount: getRandomInt(1, 4),
      features: getRandomArr(FEATURES, getRandomInt(0, FEATURES.length)),
      host: {
        name: host.name,
        avatar: host.avatar,
        isPro: getRandomBool(),
      },
      images: getRandomArr(IMAGES, getRandomInt(1, IMAGES.length)),
      description: getRandomArr(DESCRIPTIONS, getRandomInt(1, DESCRIPTIONS.length)).join(`/n`),
      reviews: generateReviews(getRandomInt(1, 5)),
      isPremium: getRandomBool(),
      isFavorite: getRandomBool(),
      coords: {
        latitude: coord[0],
        longitude: coord[1],
      },
    };
  });
};

export default () => cities.map(generateOffer).reduce((offers, item) => [...offers, ...item], []);
