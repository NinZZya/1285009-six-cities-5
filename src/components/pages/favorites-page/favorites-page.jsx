import React from 'react';
import {connect} from 'react-redux';
import PageContainer from '@/components/page-container/page-container';
import Container from '@/components/container/container';
import FavoritesList from '@/components/favorites-list/favorites-list';
import NoFavorites from '@/components/no-favorites/no-favorites';
import LoadingData from '@/components/loading-data/loading-data';
import * as OffersSelector from '@/reducer/offers/offers-selectors';
import * as CitiesSelector from '@/reducer/cities/cities-selectors';
import {DataStatus} from '@/const';
import * as Type from '@/types';


const ContainerType = {
  PAGE: `favorites`,
  FAVORITES: `page__favorites`,
};

const FavoritesPage = (props) => {
  const {favorites, offersStatus, activeCityId} = props;
  const favoritesCount = Object.keys(favorites).length;
  const isEmpty = offersStatus === DataStatus.SUCCESS && !favoritesCount;
  const emptyContent = isEmpty ? <NoFavorites /> : null;

  const favoritesContent = offersStatus === DataStatus.SUCCESS && favoritesCount ?
    <FavoritesList favorites={favorites} activeCityId={activeCityId} /> :
    null;

  return (
    <PageContainer type={ContainerType.PAGE}>
      <Container type={ContainerType.FAVORITES}>
        <section className={`favorites ${isEmpty ? `favorites--empty` : ``}`}>
          <h1 className={isEmpty ? `visually-hidden` : `favorites__title`}>
            {isEmpty ? `Favorites (empty)` : `Saved listing`}
          </h1>
          <LoadingData status={offersStatus} />
          {emptyContent}
          {favoritesContent}
        </section>
      </Container>
    </PageContainer>
  );
};


FavoritesPage.propTypes = {
  offersStatus: Type.DATA_STATUS,
  favorites: Type.FAVORITES_OFFERS,
  activeCityId: Type.ID,
};

const mapStateToProps = (state) => ({
  offersStatus: OffersSelector.getOffersStatus(state),
  activeCityId: CitiesSelector.getActiveCityId(state),
  favorites: OffersSelector.getFavoritesOffers(state),
});


export {FavoritesPage};
export default connect(mapStateToProps)(FavoritesPage);
