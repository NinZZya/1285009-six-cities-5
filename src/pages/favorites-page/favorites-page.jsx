import React from 'react';
import {connect} from 'react-redux';
import PageContainer from '../../components/page-container/page-container';
import Container from '../../components/container/container';
import FavoritesList from '../../components/favorites-list/favorites-list';
import NoFavorites from '../../components/no-favorites/no-favorites';
import Message from '../../components/message/message';
import {getActiveCityId, getFavoritesOffers, getOffersStatus} from '../../redux/offers/offers-selectors';
import {DataStatus, LOADING_MESSAGE} from '../../const';
import * as Type from '../../types';


const ContainerType = {
  PAGE: `favorites`,
  FAVORITES: `page__favorites`,
};

const FavoritesPage = (props) => {
  const {favorites, offersStatus, activeCityId} = props;
  const citiesCount = Object.keys(favorites).length;

  const loader = offersStatus === DataStatus.LOADING ?
    <Message title={LOADING_MESSAGE} /> :
    null;

  const isEmpty = offersStatus === DataStatus.SUCCESS && !citiesCount;
  const emptyContent = isEmpty ? <NoFavorites /> : null;

  const favoritesContent = offersStatus === DataStatus.SUCCESS && citiesCount ?
    <FavoritesList favorites={favorites} activeCityId={activeCityId} /> :
    null;

  return (
    <PageContainer type={ContainerType.PAGE}>
      <Container type={ContainerType.FAVORITES}>
        <section className={`favorites ${isEmpty ? `favorites--empty` : ``}`}>
          <h1 className={isEmpty ? `visually-hidden` : `favorites__title`}>
            {isEmpty ? `Favorites (empty)` : `Saved listing`}
          </h1>
          {loader}
          {emptyContent}
          {favoritesContent}
        </section>
      </Container>
    </PageContainer>
  );
};


FavoritesPage.propTypes = {
  offersStatus: Type.OFFERS_STATUS,
  favorites: Type.FAVORITES_OFFERS,
  activeCityId: Type.ID,
};

const mapStateToProps = (state) => ({
  offersStatus: getOffersStatus(state),
  activeCityId: getActiveCityId(state),
  favorites: getFavoritesOffers(state),
});


export {FavoritesPage};
export default connect(mapStateToProps)(FavoritesPage);
