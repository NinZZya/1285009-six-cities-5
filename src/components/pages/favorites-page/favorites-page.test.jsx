import React from 'react';
import {Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {FavoritesPage} from './favorites-page';
import history from '../../../constants/history';
import {DataStatus} from '../../../constants/const';
import {favoritesOffers} from '../../../mocks/offers';


describe(`Should FavoritesPage render correctly`, () => {
  it(`Should FavoritesPage render correctly loading`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <FavoritesPage
              offersStatus={DataStatus.LOADING}
            />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should FavoritesPage render correctly error`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <FavoritesPage
              offersStatus={DataStatus.ERROR}
            />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should FavoritesPage render correctly success with offers`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <FavoritesPage
              offersStatus={DataStatus.ERROR}
              favorites={favoritesOffers}
            />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
