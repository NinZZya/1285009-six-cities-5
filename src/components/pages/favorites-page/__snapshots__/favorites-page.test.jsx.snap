// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoritesPage render correctly Should FavoritesPage render correctly error 1`] = `
<main
  className="page__main page__main--favorites "
>
  <div
    className="container page__favorites-container "
  >
    <section
      className="favorites "
    >
      <h1
        className="favorites__title"
      >
        Saved listing
      </h1>
      <div
        style={
          Object {
            "alignItems": "center",
            "color": "#ff0000",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "margin": "20px",
            "padding": "20px",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <h2
          style={
            Object {
              "fontSize": "24px",
              "fontStyle": "oblique",
              "fontWeight": 700,
              "lineHeight": 1.1667,
            }
          }
        >
          Error
        </h2>
        <p>
          Error loading data, try again later...
        </p>
      </div>
    </section>
  </div>
</main>
`;

exports[`Should FavoritesPage render correctly Should FavoritesPage render correctly loading 1`] = `
<main
  className="page__main page__main--favorites "
>
  <div
    className="container page__favorites-container "
  >
    <section
      className="favorites "
    >
      <h1
        className="favorites__title"
      >
        Saved listing
      </h1>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "fontSize": "38px",
            "fontStyle": "oblique",
            "fontWeight": 700,
            "justifyContent": "flex-start",
            "lineHeight": 0.7,
            "margin": "20px",
            "padding": "30px",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <h3
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "space-between",
            }
          }
        >
          <img
            alt="logo"
            height="50"
            src="/img/logo.svg"
            style={
              Object {
                "marginRight": "20px",
              }
            }
            width="100"
          />
          <svg
            height="100%"
            version="1.1"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <path
              id="path"
            >
              <animate
                attributeName="d"
                begin="0s"
                dur="3s"
                from="m0,110 h0"
                repeatCount="indefinite"
                to="m0,110 h1100"
              />
            </path>
            <text
              fill="black"
            >
              <textPath
                xlinkHref="#path"
              >
                Loading...
              </textPath>
            </text>
          </svg>
        </h3>
      </div>
    </section>
  </div>
</main>
`;

exports[`Should FavoritesPage render correctly Should FavoritesPage render correctly success with offers 1`] = `
<main
  className="page__main page__main--favorites "
>
  <div
    className="container page__favorites-container "
  >
    <section
      className="favorites "
    >
      <h1
        className="favorites__title"
      >
        Saved listing
      </h1>
      <div
        style={
          Object {
            "alignItems": "center",
            "color": "#ff0000",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "margin": "20px",
            "padding": "20px",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <h2
          style={
            Object {
              "fontSize": "24px",
              "fontStyle": "oblique",
              "fontWeight": 700,
              "lineHeight": 1.1667,
            }
          }
        >
          Error
        </h2>
        <p>
          Error loading data, try again later...
        </p>
      </div>
    </section>
  </div>
</main>
`;
